internal/infoblox/paging_ibclient.go

```diff
@@ -0,0 +1,60 @@
+package infoblox
+
+import (
+	"fmt"
+	"reflect"
+
+	ibclient "github.com/infobloxopen/infoblox-go-client/v2"
+)
+
+func PagingGetObject[T any](
+	c ibclient.IBConnector,
+	obj ibclient.IBObject,
+	ref string,
+	queryParams map[string]string,
+	res *[]T,
+) (err error) {
+
+	pagingResponse := pagingResponseStruct[T]{
+		NextPageId: "",
+		Result:     make([]T, 0),
+	}
+
+	//copy query params and update them
+	queryParamsCopy := map[string]string{}
+	for k, v := range queryParams {
+		queryParamsCopy[k] = v
+	}
+
+	queryParamsCopy["_return_as_object"] = "1"
+	queryParamsCopy["_paging"] = "1"
+	queryParamsCopy["_max_results"] = "1000"
```

Comment:
I was testing against IBX with `_max_results` "50", "100" but without effect on paging. Is tehere any minimum page size `c.GetObject` retrieves ? is `_max_results` correct name ?

Reply:
_max_results is the correct key https://ipam.illinois.edu/wapidoc/index.html#get but I think it should only override the value if one isn't set not set it to 1000 for everything; I'll change it.
The min value is 1, and speeds up queries if you're only looking for 1 thing; it will stop at the first match. This is the best wapi doc I could find https://insights.infoblox.com/resources-deployment-guides/infoblox-deployment-infoblox-rest-api